packages/lib-classifier/src/components/Classifier/components/MetaTools/components/HidePreviousMarksButton/HidePreviousMarksButton.js

```diff
@@ -1,23 +1,38 @@
 import { MetaToolsButton } from '@zooniverse/react-components'
 import counterpart from 'counterpart'
 import PropTypes from 'prop-types'
-import { FormView, FormViewHide } from 'grommet-icons'
+import { FormView, FormViewHide, Hide } from 'grommet-icons'
 import React from 'react'
 
+import SHOWN_MARKS from '../../../../../../helpers/shownMarks'
 import en from './locales/en'
 
 counterpart.registerTranslations('en', en)
 
 export default function HidePreviousMarksButton (props) {
-  const { disabled, hidePreviousMarks, onClick } = props
-  const icon = hidePreviousMarks ? <FormViewHide /> : <FormView />
-  const text = hidePreviousMarks ?
-    counterpart('HidePreviousMarksButton.show') :
-    counterpart('HidePreviousMarksButton.hide')
+  const { disabled, onClick, shownMarks, type } = props
+  let text, icon;
+  const isDrawingTask = type === 'drawing'
+
+  switch(shownMarks) {
+    case SHOWN_MARKS.ALL:
+      text = isDrawingTask ? counterpart('HidePreviousMarksDrawingButton.hide')
+        : counterpart('HidePreviousMarksTranscriptionButton.showUser')
+      icon = <FormView />
+      break;
+    case SHOWN_MARKS.USER:
+      text = counterpart('HidePreviousMarksTranscriptionButton.hide')
+      icon = <Hide />
+      break;
+    default:
+      text = isDrawingTask ? counterpart('HidePreviousMarksDrawingButton.show')
+        : counterpart('HidePreviousMarksTranscriptionButton.show')
+      icon = <FormViewHide />
+  }
 
   return (
     <MetaToolsButton
-      aria-checked={hidePreviousMarks}
+      aria-checked={shownMarks}
       disabled={disabled}
       icon={icon}
       role='checkbox'
```

Comment:
Now that there are multiple states to choose from, this is emulating a select menu or radio group, rather than a checkbox.

Reply:
For accessibility, is the best way to go about this to use `role='menu'` and then remove `aria-checked` in favor of... `aria-label`? I would think the aria item should tell what the current state is: "All Marks", "User Marks", "No Marks".

Reply:
Buttons can’t have the `menu` role, because a menu has to wrap a list of `menuitem` roles.

The button label should tell you what the button does when you press it. I don’t think anyone will understand the label as being the current state. That sounds confusing. 

Button state can be either `pressed` or `not-pressed`. There’s also a `switch` role but that’s either on or off. 

I’m having trouble finding examples of buttons that have more than two states. The closest I can think of would be a group of 3 radio buttons, one of which is selected. However, the expected interaction for radio buttons would be that you change the selected state with the cursor keys.

Reply:
It looks like there is also an [indeterminate state](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#Indeterminate_state_checkboxes) for checkboxes that can be set via Javascript. I can try using this with the label being "Show All Marks," "Show User Marks," and "Hide All Marks."

Reply:
Actually, I'm starting to think of reasons this would still conflict with the label. I'll have to think of another solution.